lenskit-core/src/main/java/org/grouplens/lenskit/data/source/TextDataSource.java

```diff
@@ -181,6 +182,7 @@ public static TextDataSource fromSpec(TextDataSourceSpec spec) {
         bld.setFormat(fmt);
         bld.setItemFile(spec.getItemFile());
         bld.setItemNameFile(spec.getItemNameFile());
+        bld.setHeaderLines(fmt.getHeaderLines());
```

Comment:
Not quite.  This should be:

    fmt.setHeaderLines(spec.getHeaderLines());

And should probably be higher up, where you're setting up the format.